lib/app/app.rb

```diff
@@ -54,8 +57,6 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   before do
-    @connector = settings.connector_class.new
```

Comment:
Was there a reason to get rid of this initialization in the before block?

Reply:
So it was a chain of thoughts.

JobRunner actually creates a new thread that does the syncing logic. A good idea would be to create an instance of connector inside the thread. Therefore, we don't want to create an instance of connector before running `/documents` endpoint.

Further, if the controller becomes async, it feels like objects with Per-Request lifecycle should be created inside of each individual request, not use a shared `@connector` instance variable that can be overwritten by a different request/thread - I might be wrong here though, I'm not super familiar with how instance variables are handled when multiple requests hit Sinatra server at the same time.

Reply:
The way you have it is probably fine. For what it's worth, I do not believe separate requests share `@connector`.